extensions/panache/rest-data-panache/deployment/src/main/java/io/quarkus/rest/data/panache/deployment/utils/ResourceMethodListenerUtils.java

```diff
@@ -0,0 +1,49 @@
+package io.quarkus.rest.data.panache.deployment.utils;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.jboss.jandex.ClassInfo;
+
+import io.quarkus.rest.data.panache.deployment.ResourceMethodListenerBuildItem;
+
+public final class ResourceMethodListenerUtils {
+
+    private ResourceMethodListenerUtils() {
+
+    }
+
+    public static List<ClassInfo> getListenersByEntityType(List<ResourceMethodListenerBuildItem> resourceMethodListeners,
+            String entityTypeName) {
+        Class<?> entityClass = loadClass(entityTypeName);
```

Comment:
I loading classes in the deployment is generally frowned upon, though I forgot why exactly (performance? problems with the deployment classloader?).

Anyway, can you use the Jandex index instead? You should be able to retrieve a `ClassInfo` from `CombinedIndexBuildItem`, then retrieve all superclasses and superinterfaces from that `ClassInfo`. We do that in several places already, e.g. https://github.com/quarkusio/quarkus/blob/a23639025a26b3f4073efd64a5188191ca0067f6/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/JpaJandexScavenger.java#L451-L456

Reply:
Ok ok, I'm not sure if there is a performance benefit, but for the deployment extensions, the Index is generally used, so I've already changed it. Many thanks for checking!